controllers/grid/files/proof/ManageProofFilesGridHandler.inc.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * @file controllers/grid/files/proof/ManageProofFilesGridHandler.inc.php
+ *
+ * Copyright (c) 2014-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ManageProofFilesGridHandler
+ * @ingroup controllers_grid_files_proof
+ *
+ * @brief Handle the editor's proof files selection grid (selects which files to include)
+ */
+
+import('lib.pkp.controllers.grid.files.SelectableSubmissionFileListCategoryGridHandler');
+
+class ManageProofFilesGridHandler extends SelectableSubmissionFileListCategoryGridHandler {
+	/**
+	 * Constructor
+	 */
+	function ManageProofFilesGridHandler() {
+		import('controllers.grid.files.proof.ProofFilesCategoryGridDataProvider');
+		parent::SelectableSubmissionFileListCategoryGridHandler(
+			new ProofFilesCategoryGridDataProvider(),
+			WORKFLOW_STAGE_ID_PRODUCTION
+		);
+
+		$this->addRoleAssignment(
+			array(ROLE_ID_SUB_EDITOR, ROLE_ID_MANAGER),
+			array(
+				'fetchGrid', 'fetchCategory', 'fetchRow',
+				'addFile', 'downloadFile', 'deleteFile',
+				'updateProofFiles',
+			)
+		);
+
+		// Set the grid title.
+		$this->setTitle('submission.pageProofs');
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
```

Comment:
No, actually -- without the submission access policy the publication format required policy won't work, because the submission won't be in the authorized context yet.